### DISCUSSION TITLE
intermittent "No Data" in Graphana

### DISCUSSION THREAD

**Discussion by ejg3855** [David00/rpi-power-monitor#116] [Q&A] [answered]
I am getting a consistent NO DATA message on my CT2 data line, but its inconsistent. One refresh of the data there will be a value there and in the next refresh it might say NO DATA. The oddity is when I use the default dashboard for the CT probe data I do not get any drops of the data. 

Im not sure how to go about trouble shooting this as I believe the data stream is solid based upon the default CT probe data dashboard. But when I bring it into where and how I want to display it, chaos happens.

`{`
  "request": {
    "url": "api/ds/query?ds_type=__expr__&expression=true&requestId=Q340",
    "method": "POST",
    "data": {
      "queries": [
        {
          "alias": "$tag_status",
          "datasource": {
            "uid": "PB39459CF186E3D64",
            "type": "influxdb"
          },
          "groupBy": [
            {
              "params": [
                "status"
              ],
              "type": "tag"
            }
          ],
          "hide": true,
          "measurement": "raw_cts",
          "orderByTime": "ASC",
          "policy": "default",
          "query": "SELECT \"power\" FROM \"net\" WHERE $timeFilter",
          "rawQuery": false,
          "refId": "A",
          "resultFormat": "table",
          "select": [
            [
              {
                "params": [
                  "power"
                ],
                "type": "field"
              }
            ]
          ],
          "tags": [
            {
              "key": "ct",
              "operator": "=",
              "value": "1"
            }
          ],
          "adhocFilters": [],
          "limit": "",
          "slimit": "",
          "tz": "",
          "datasourceId": 1,
          "intervalMs": 50,
          "maxDataPoints": 100
        },
        {
          "alias": "$tag_status",
          "datasource": {
            "uid": "PB39459CF186E3D64",
            "type": "influxdb"
          },
          "groupBy": [
            {
              "params": [
                "status"
              ],
              "type": "tag"
            }
          ],
          "hide": true,
          "measurement": "raw_cts",
          "orderByTime": "ASC",
          "policy": "default",
          "query": "SELECT \"power\" FROM \"net\" WHERE $timeFilter",
          "rawQuery": false,
          "refId": "B",
          "resultFormat": "table",
          "select": [
            [
              {
                "params": [
                  "power"
                ],
                "type": "field"
              },
              {
                "params": [
                  "3"
                ],
                "type": "bottom"
              }
            ]
          ],
          "tags": [
            {
              "key": "ct",
              "operator": "=",
              "value": "2"
            }
          ],
          "adhocFilters": [],
          "limit": "",
          "slimit": "",
          "tz": "",
          "datasourceId": 1,
          "intervalMs": 50,
          "maxDataPoints": 100
        },
        {
          "datasource": {
            "type": "__expr__",
            "uid": "__expr__",
            "name": "Expression"
          },
          "expression": "($A+$B)",
          "hide": false,
          "refId": "House Active",
          "type": "math",
          "window": ""
        }
      ],
      "range": {
        "from": "2023-11-13T19:30:16.435Z",
        "to": "2023-11-13T19:30:21.435Z",
        "raw": {
          "from": "now-5s",
          "to": "now"
        }
      },
      "from": "1699903816435",
      "to": "1699903821435"
    },
    "hideFromInspector": false
  },
  "response": {
    "results": {
      "House Active": {
        "status": 200,
        "frames": [
          {
            "schema": {
              "refId": "House Active",
              "meta": {
                "type": "timeseries-multi",
                "typeVersion": [
                  0,
                  1
                ]
              },
              "fields": [
                {
                  "name": "Time",
                  "type": "time",
                  "typeInfo": {
                    "frame": "time.Time"
                  }
                },
                {
                  "name": "House Active",
                  "type": "number",
                  "typeInfo": {
                    "frame": "float64",
                    "nullable": true
                  },
                  "labels": {
                    "status": ""
                  }
                }
              ]
            },
            "data": {
              "values": [
                [
                  1699903816942,
                  1699903819348
                ],
                [
                  1565.1880186647345,
                  1559.4110116204251
                ]
              ]
            }
          }
        ],
        "refId": "House Active"
      }
    }
  }
}
`

**Comment by David00** [comment]
Hi @ejg3855, can you elaborate on this part?

> But when I bring it into where and how I want to display it, chaos happens

Where are you trying to bring the data into? It looks like you're trying to use the InfluxDB API which I'm not super familiar with.

**Reply by David00** [reply, marked as answer]
Looking a bit more closely at your API request, it could just be that the time range is too short:

```
"range": {
"from": "2023-11-13T19:30:16.435Z",
"to": "2023-11-13T19:30:21.435Z",
"raw": {
"from": "now-5s",
"to": "now"
}
```

I'm referring to the `now-5s` part specifically. The power monitor takes readings quicker than that, but it actually caches 25 readings and then writes them as a batch to reduce overheard and wear and tear on the microSD card. So, the actual database might not have any update for about 20-30 seconds, but when it gets updated, that 20-30 second gap will get filled in.  

Can you try increasing the time range to `now-30s`? That will give you multiple results, so you might also want to try changing your query to only return the last value:

`"SELECT last("power") FROM "net" WHERE $timeFilter"`

**Comment by ejg3855** [comment]
I just copied and edited your CT probes dashboard.
